src/main/java/uk/gov/register/presentation/resource/SearchResource.java

```diff
@@ -42,7 +42,7 @@ public Response findByPrimaryKey(@PathParam("primaryKey") String key, @PathParam
             return buildResponse(new SingleResultView("/templates/entry.mustache", entry.isPresent() ? entry.get() : null));
         }
 
-        throw new BadRequestException("Key: " + key + " is not primary key of the register.");
+        throw new NotFoundException();
```

Comment:
I disagree. We don't define a resource `/{key}/{value}` and require that `{key}` = `postcode`, we define a resource `/postcode/{value}` and leave everything else undefined.

Think of it from the point of view of the consumer, who knows nothing about the back-end implementation.  They know that `school.openregister.org/hash/{value}` is defined, and that `school.openregister.org/school/{value}` is defined, but there's nothing documented leading them to expect anything of `school.openregister.org/{anythingelse}/{value}`, so it's natural for it to be a 404.